astropy/uncertainty/tests/test_distribution.py

```diff
@@ -15,30 +17,42 @@
     SMAD_FACTOR = 1 / norm.ppf(0.75)
 
 
-def test_numpy_init():
-    # Test that we can initialize directly from a Numpy array
-    rates = np.array([1, 5, 30, 400])[:, np.newaxis]
-    parr = np.random.poisson(rates, (4, 1000))
-    Distribution(parr)
-
-
-def test_numpy_init_T():
-    rates = np.array([1, 5, 30, 400])
-    parr = np.random.poisson(rates, (1000, 4))
-    Distribution(parr.T)
-
-
-def test_quantity_init():
-    # Test that we can initialize directly from a Quantity
-    pq = np.random.poisson(np.array([1, 5, 30, 400])[:, np.newaxis],
-                           (4, 1000)) * u.ct
-    Distribution(pq)
-
-
-def test_quantity_init_T():
-    # Test that we can initialize directly from a Quantity
-    pq = np.random.poisson(np.array([1, 5, 30, 400]), (1000, 4)) * u.ct
-    Distribution(pq.T)
+class TestInit:
+    @classmethod
+    def setup_class(self):
+        self.rates = np.array([1, 5, 30, 400])[:, np.newaxis]
+        self.parr = np.random.poisson(self.rates, (4, 1000))
+        self.parr_t = np.random.poisson(self.rates.squeeze(), (1000, 4))
+
+    def test_numpy_init(self):
+        # Test that we can initialize directly from a Numpy array
+        Distribution(self.parr)
+
+    def test_numpy_init_T(self):
+        Distribution(self.parr_t.T)
+
+    def test_quantity_init(self):
+        # Test that we can initialize directly from a Quantity
+        pq = self.parr << u.ct
+        pqd = Distribution(pq)
+        assert isinstance(pqd, u.Quantity)
+        assert isinstance(pqd, Distribution)
+        assert isinstance(pqd.value, Distribution)
+        assert_array_equal(pqd.value.distribution, self.parr)
+
+    def test_quantity_init_T(self):
+        # Test that we can initialize directly from a Quantity
+        pq = self.parr_t << u.ct
+        Distribution(pq.T)
+
+    def test_quantity_init_with_distribution(self):
+        # Test that we can initialize a Quantity from a Distribution.
+        pd = Distribution(self.parr)
+        qpd = pd << u.ct
```

Comment:
This is really smooth! As an editorial note I can't help but think we need to get the `<<` syntax more widely known...

Reply:
Indeed!  Though I think the multiplication remains more intuitive and for most purposes the extra copy really doesn't matter.

Reply:
does ``pd <<= u.ct`` work in these instances to avoid a copy?

Reply:
`pd << u.ct` already does not make a copy (it calls `Quantity(pd, u.ct, copy=False)`). `pd <<= u.ct` does not do anything different, really, except that `pd` now points to the new `QuantityDistribution` object.